### PR TITLE
Add Access List documentation.

### PR DESCRIPTION
**Pull request by mdwn**
Access List documentation has been added, providing a reference for Access Lists for Teleport v14.

### CODE REVIEW

File: docs/pages/access-controls/access-lists.mdx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+title: Access Lists (Preview)
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists.mdx:2]: @r0mant I thought we were getting rid of preview with minor exceptions for things like assist. Should we remove it here?

File: docs/pages/access-controls/access-lists.mdx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+description: Use and enforce trusted devices with Teleport
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists.mdx:3]: We need to do a better job at not leaving copy-pasta behind. This is not a device trust guide.

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+---
+title: A Simple Access List Example
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists/guide.mdx:2]: Should this be called getting started instead?

File: docs/pages/access-controls/access-lists/reference.mdx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+Access Lists allow Teleport users to be granted long term access to resources
+managed within Teleport. With Access Lists, administrators and access list
+owners can regularly audit and control membership to specific roles and
+traits, which then tie easily back into Teleport's existing RBAC system.
+
+## What do Access Lists do?
+
+Access Lists grant roles and traits to users on a long lived basis. Users who are added
+to Access Lists as members and meet membership requirements will be granted these roles
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists/reference.mdx:18]:
```suggestion
to Access Lists and meet membership requirements will be granted these roles
```

File: docs/pages/access-controls/access-lists/reference.mdx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+and traits upon login in Teleport. By combining this functionality with an auditing
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists/reference.mdx:19]:
```suggestion
and traits when they sign in to Teleport. By combining this functionality with an auditing
```

Comment (zmb3) [inline docs/pages/access-controls/access-lists/reference.mdx:19]: What is "an auditing functionality"

File: docs/pages/access-controls/access-lists/reference.mdx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+Access Lists are intended for longer lived access to Teleport resources and Access Requests
+are intended for temporary elevation of privileges. Access conveyed by an Access List is
+expected to live on the order of months and access conveyed by an Access Request is intended
+to live on the order of hours or days.
+
+## Access List Ownership
+
+Access List owners are Teleport users who are granted special privileges over
+an Access List. These owners are defined explicitly as part of the Access List, and
+must be added by a Teleport administrator. Owners must meet requirements in order for
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists/reference.mdx:34]: Can we clarify who can add owners? There is not a formal concept of an "administrator" in Teleport.

File: docs/pages/access-controls/access-lists/reference.mdx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  # with the next audit date. If an Access List is not audited by the next audit date, the
+  # roles and traits granted by this Access List will be revoked from the users until the
+  # audit occurs.
+  audit:
+    # Audit frequency defaults to 6 months. Format to golang's time.ParseDuration function:
+    # https://pkg.go.dev/time#ParseDuration
+    frequency: 4380h
+    # The next time this Access List must be audited by.
+    next_audit_date: "2024-01-01T00:00:00Z"
+  description: "A plaintext description of the Access List and its purpose"
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists/reference.mdx:84]:
```suggestion
  description: "A description of the Access List and its purpose"
```

File: docs/config.json [hunk 8]
```diff
@@ -374,6 +374,23 @@
             }
           ]
         },
+        {
+          "title": "Access Lists",
+          "slug": "/access-controls/access-lists/",
+          "forScopes": ["enterprise", "cloud"],
+          "entries": [
+            {
+              "title": "A Simple Access List Example",
```
Comment (r0mant) [inline docs/config.json:383]: Feels like this title should be something else.

File: docs/pages/access-controls/access-lists.mdx [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+layout: tocless-doc
+---
+
+<Admonition type="warning">
+Access Lists are currently in Preview mode.
+</Admonition>
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists.mdx:9]:
```suggestion
```

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+description: Learn how to use Access Lists to manage and audit long lived access to Teleport managed resources.
+---
+
+<Admonition type="warning">
+Access Lists are currently in Preview mode.
+</Admonition>
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/guide.mdx:8]:
```suggestion
```

File: docs/pages/access-controls/access-lists/reference.mdx [hunk 11]
```diff
@@ -0,0 +1,122 @@
+---
+title: Access List Reference
+description: An explanation and overview of Access Lists in Teleport.
+---
+
+<Admonition type="warning">
+Access Lists are currently in Preview mode.
+</Admonition>
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/reference.mdx:8]:
```suggestion
```

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+This guide will help you set up a basic Access List.
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/guide.mdx:10]:
```suggestion
This guide will help you:
- Create an access list
- Assign a member to it
- Verify permissions granted through the list membership
```

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+- An administrative user with the preset `editor` role, which will have permissions to create Access Lists.
+
+## Step 1/3. Create a test user
+
+First, we need to create a simple test user that has only the `requester` role, which has no default access
+to anything within a cluster. This user will only be used for the purposes of this guide, so you may use
+another user if you so choose.
+
+```code
+$ tctl users add test-user --roles=requester --logins=root,ubuntu,ec2-user
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/guide.mdx:29]: Can we replace this with the instructions of inviting the user using web UI?

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+```
+
+Then, follow the link provided and create credentials for this user. Try logging in to verify that no resources
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/guide.mdx:32]: Will the login work if you only have requester role?

File: docs/pages/access-controls/access-lists/reference.mdx [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+      required_trait1: required_value1
+```
+
+## Managing Access Lists from the CLI
+
+In addition to using the web UI, Access Lists can be created and managed from the CLI
+as well. To create an access list from the CLI, create an Access List YAML file (as described
+above) and run `tctl create <filename>`. Access Lists can be updated by using `tctl create -f <filename>`.
+
+`tctl` also supports a subset of Access List focused commands under the `tctl acl` subcommand.
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/reference.mdx:120]: Give a link to the CLI reference page somewhere here.

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+We'll leave the the requirements fields blank and then click "Create Access List" at the bottom
+of the page.
+
+## Step 3/3. Login as the test user
+
+Again, login as the test user. When logging in now, you should now see the resources contained within
+the cluster, and should be able to interact with them as expected.
+
+## Next Steps
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/guide.mdx:64]: We should add a link to Access Lists to the "next steps" for access requests guides too.

**Review by r0mant — COMMENTED** [review]
First quick pass, I'll take another deeper look as well.

File: docs/pages/access-controls/access-lists/guide.mdx (hunk 14 above)
Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx:32]: Yes it will.

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+- Assign a member to it
+- Verify permissions granted through the list membership
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+- A running Teleport cluster with some built in resources. Following the
+  [Web Application Access Guide](../../application-access/guides/connecting-apps.mdx) will create a base with a
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists/guide.mdx:17]: The guide just tells you that you need some resources and kind of assumes that you don't have access to the resource before being added to the access list and that you do have access after being added. I wonder if there's a way to make this more concrete.

Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx:17]: I'm sure there is. What do you think about making the `Connecting Apps` guide a little more fully featured in this doc? We effectively need to instruct the user, "create a few protected resources" somehow.

Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx]: I've added in a step to add the dumper app to the cluster. Let me know what you think.

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+![Create Test User](../../../img/access-controls/access-lists/create-test-user.png)
+
+Click "Save," and then navigate to the provided URL in order to set up the credentials for your test user.
+Try logging into the cluster with the test user to verify that no resources show up in the resources page.
+
+## Step 3/4. Create an Access List
+
+Next, we'll create a simple access list that will grant the `access` role to its members.
+Login as the administrative user mentioned in the prerequisites. Navigate to the management pane and
+click on access lists. Click on "Create an Access List."
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/guide.mdx:53]:
```suggestion
click on access lists. Click on "Create an Access List".
```

File: docs/pages/access-controls/access-lists/reference.mdx [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+version: v1
+kind: access_list
+metadata:
+  name: ea6cccbe-ceac-4776-8a89-4b1365fc03f5
+spec:
+  title: "Access List Title"
+  # audit defines how frequently an Access List and its membership must be audited, along
+  # with the next audit date. If an Access List is not audited by the next audit date, the
+  # roles and traits granted by this Access List will be revoked from the users until the
+  # audit occurs.
```
Comment (r0mant) [inline docs/pages/access-controls/access-lists/reference.mdx:73]: Let's remove this note for now, until we iron out what kind of behavior we want.
```suggestion
  # audit defines how frequently an Access List and its membership must be audited, along
  # with the next audit date.
```

**Review by r0mant — APPROVED** [review]

File: docs/pages/access-controls/access-lists/guide.mdx (hunk 18 above)
Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx:53]: It's actually considered more proper to put the period inside the quotes. https://www.hamilton.edu/academics/centers/writing/style/essentials/punctuation-of-quotations#:~:text=The%20final%20period%20or%20comma,the%20period%20follows%20the%20citation.

Comment (r0mant) [reply docs/pages/access-controls/access-lists/guide.mdx:53]: Huh, well, I can't argue with an .edu link 😄

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+
+![Navigate to create new Access List](../../../img/access-controls/access-lists/create-new-access-list.png)
+
+Here, fill in a title, description, and grant the `access` role.
+
+![Fill out Access List Fields](../../../img/access-controls/access-lists/fill-out-access-list-fields.png)
```
[the remark this answers is not in the document]
Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx:59]: I added this to `reference.mdx`. Let me know what you think.

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+
+Under "List Owners" select `editor` as a required role, then add your administrative user under "Add
```
Comment (zmb3) [inline docs/pages/access-controls/access-lists/guide.mdx:61]: Can we describe what making `editor` a required role actually does? I assume it means anyone with `editor` can manage the access list?

**Review by zmb3 — APPROVED** [review]

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+This guide will help you:
+- Create an access list
+- Assign a member to it
+- Verify permissions granted through the list membership
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
```
[the remark this answers is not in the document]
Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx:15]: Yes it does.

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 23]
```diff
@@ -0,0 +1,80 @@
...
+Under "List Owners" select `editor` as a required role, then add your administrative user under "Add
+Eligible List Owners."
+
+![Select an owner](../../../img/access-controls/access-lists/select-owner.png)
+
+Under "Members" select `requester` as a required role, then add your test user to the access list.
+
+![Add a member](../../../img/access-controls/access-lists/add-member.png)
+
+We'll leave the the requirements fields blank and then click "Create Access List" at the bottom
```
[the remark this answers is not in the document]
Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx:70]: This actually doesn't apply any more, so I'll excise this.

File: docs/pages/access-controls/access-lists/guide.mdx (hunk 21 above)
Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx:62]: I've described the requirements here and in membership.

File: docs/pages/access-controls/access-lists/guide.mdx [hunk 24]
```diff
@@ -0,0 +1,80 @@
...
+Here, fill in a title, description, and grant the `access` role.
+
+![Fill out Access List Fields](../../../img/access-controls/access-lists/fill-out-access-list-fields.png)
+
+Under "List Owners" select `editor` as a required role, then add your administrative user under "Add
+Eligible List Owners."
+
+![Select an owner](../../../img/access-controls/access-lists/select-owner.png)
+
+Under "Members" select `requester` as a required role, then add your test user to the access list.
```
[the remark this answers is not in the document]
Comment (mdwn) [reply docs/pages/access-controls/access-lists/guide.mdx]: I've clarified this.

**Pull request merged**